packages/hoppscotch-backend/src/user-collection/user-collection.service.ts

```diff
@@ -1087,6 +1089,17 @@ export class UserCollectionService {
       ),
     );
 
+    if (!isCollectionDuplication) {
+      const collectionData = await this.fetchCollectionData(
+        userCollections[0].id,
+      );
+      if (!E.isLeft(collectionData))
+        this.pubsub.publish(
+          `user_coll/${userID}/duplicated`,
+          collectionData.right,
+        );
+    }
+
     userCollections.forEach((collection) =>
       this.pubsub.publish(`user_coll/${userID}/created`, this.cast(collection)),
     );
```

Comment:
We require only the `.../duplicated` topic to be published during a duplicate action. Also, shouldn't the check be the other way around where the `.../duplicated` topic is published if `isCollectionDuplication` is set to `true`?

```suggestion
    if (isCollectionDuplication) {
      const collectionData = await this.fetchCollectionData(
        userCollections[0].id,
      );

      if (E.isRight(collectionData))
        this.pubsub.publish(
          `user_coll/${userID}/duplicated`,
          collectionData.right,
        );
    } else {
      userCollections.forEach((collection) =>
        this.pubsub.publish(
          `user_coll/${userID}/created`,
          this.cast(collection),
        ),
      );
    }
```